### ISSUE TITLE
Question: Is there a way to validate lwm2m object definition in json format?

### ISSUE THREAD

**Issue by mofosyne** [eclipse/leshan#492]
I am looking for a json schema that is similar to http://openmobilealliance.org/tech/profiles/LWM2M.xsd but used for json instead of xml object definitions.

But I can't seem to find it. It would be used to validate that something like below is well formed

```
{
    "name": "Test Meter",
    "id": 666,
    "instancetype": "mulitple",
    "mandatory": false,
    "description": "",
    "resourcedefs": [
      {
        "id": 0,
        "name": "Pressure",
        "operations": "RW",
        "instancetype": "single",
        "mandatory": true,
        "type": "float",
        "range": "",
        "units": "",
        "description": ""
      }
    ]
  };
```

For your information, I would use the jsonschema file to prevalidate any object models before it is imported to the server. (Via a node.js script)

-----

Reference:

* Object model in java is located in `leshan-core/src/main/java/org/eclipse/leshan/core/model/ObjectModel.java`

* Object Definition in XML format is parsed in `leshan-core/src/main/java/org/eclipse/leshan/core/model/DDFFileParser.java`

* Models are loaded via `leshan-core/src/main/java/org/eclipse/leshan/core/model/ObjectLoader.java`

**Comment by sbernard31** [comment]
There is no json schema :/

The json format was introduce at the beginning because of legal issue about xml object definitions format from OMA.

Now xml files can be reused without constraints, json format is deprecated. I mean there is no needs to support another format than the one supported by OMA.

**Comment by mofosyne** [comment]
I see. Then is there a way to migrate existing json models to XML instead?

**Comment by sbernard31** [comment]
> Then is there a way to migrate existing json models to XML instead?

There is no way to do that out of the box.
But you can use `ObjectLoader.loadJsonStream(InputStream), to load JSON file then write a DDFFileSerializer (like our DDFFileParser). This should not be so hard. You can even contribute it.

The xml schema is available [here](http://www.openmobilealliance.org/wp/OMNA/LwM2M/LwM2MRegistry.html)

> Also what is the timeline before the json input format is disabled?

We do not plan to remove json input support at short/mid term (It will probably available in the 1.0.0 version) but we will not provide more feature and we can not assure that we will keep it at long term. It is clearly deprecated.

**Comment by sbernard31** [comment]
I feel we can close this issue.
